corehq/apps/accounting/invoicing.py

```diff
@@ -655,8 +632,27 @@ def unit_cost(self):
 
     @property
     def quantity(self):
+        if self.invoice.is_customer_invoice and self.invoice.account.invoicing_plan != InvoicingPlan.MONTHLY:
+            return self.num_excess_users_over_period
         return self.num_excess_users
 
+    @property
+    def num_excess_users_over_period(self):
+        # Iterate through all months in the invoice date range to aggregate total users into one line item
+        excess_users = 0
+        month_end = self.invoice.date_end
+        while month_end > self.invoice.date_start:
+            total_users = 0
+            for domain in self.subscribed_domains:
+                try:
+                    history = DomainUserHistory.objects.get(domain=domain, record_date=month_end)
+                    total_users += max(history.num_users - self.rate.monthly_limit, 0)
+                except DomainUserHistory.DoesNotExist:
+                    total_users += 0
+            excess_users += max(total_users - self.rate.monthly_limit, 0)
+            _, month_end = get_previous_month_date_range(month_end)
```

Comment:
non-blocking nit: It can be preferable where possible to generate the list of dates outside of the loop, and then pass the list into a for loop.  This way, the loop body is only focused on updating ```excess_users```, which separates out different pieces of the logic for clarity.